From 62107ba949fe70e5ae0573147d82cbb52bd1fc08 Mon Sep 17 00:00:00 2001
From: Mahesh Bodapati <mbodapat@xilinx.com>
Date: Mon, 8 Nov 2021 22:01:23 +0530
Subject: [PATCH 30/34] ld/emulparams/elf64microblaze: Fix emulation generation

Compilation fails when building ld-new with:

ldemul.o:(.data.rel+0x820): undefined reference to `ld_elf64microblazeel_emulation'
ldemul.o:(.data.rel+0x828): undefined reference to `ld_elf64microblaze_emulation'

The error appears to be that the elf64 files were referencing the elf32 emulation.

Signed-off-by: Mark Hatle <mark.hatle@xilinx.com>
---
 ld/emulparams/elf64microblaze.sh   | 2 +-
 ld/emulparams/elf64microblazeel.sh | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/ld/emulparams/elf64microblaze.sh b/ld/emulparams/elf64microblaze.sh
index 9c7b0eb7080..7b4c7c411bd 100644
--- a/ld/emulparams/elf64microblaze.sh
+++ b/ld/emulparams/elf64microblaze.sh
@@ -19,5 +19,5 @@ NOP=0x80000000
 #$@{RELOCATING+ PROVIDE (__stack = 0x7000);@}
 #OTHER_RELOCATING_SECTIONS='PROVIDE (_stack = _end + 0x1000);'
 
-TEMPLATE_NAME=elf32
+TEMPLATE_NAME=elf
 #GENERATE_SHLIB_SCRIPT=yes
diff --git a/ld/emulparams/elf64microblazeel.sh b/ld/emulparams/elf64microblazeel.sh
index 9c7b0eb7080..7b4c7c411bd 100644
--- a/ld/emulparams/elf64microblazeel.sh
+++ b/ld/emulparams/elf64microblazeel.sh
@@ -19,5 +19,5 @@ NOP=0x80000000
 #$@{RELOCATING+ PROVIDE (__stack = 0x7000);@}
 #OTHER_RELOCATING_SECTIONS='PROVIDE (_stack = _end + 0x1000);'
 
-TEMPLATE_NAME=elf32
+TEMPLATE_NAME=elf
 #GENERATE_SHLIB_SCRIPT=yes
-- 
2.37.1 (Apple Git-137.1)

